src/App.tsx

```diff
@@ -16,47 +18,108 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goodsArr, setGoodsArr] = useState(goodsFromServer);
+  const [sortValue, setSortValue] = useState('');
+  const [reverseValue, setReverseValue] = useState(false);
+  const [resetValue, setResetValue] = useState(false);
+
+  const SORT_BY_ALPHABETICALLY = 'alphabetically';
+  const SORT_BY_LENGTH = 'length';
+
+  const sortBy = (value: string) => {
+    if (value === SORT_BY_ALPHABETICALLY) {
+      if (reverseValue) {
+        setGoodsArr(prevState => [...prevState].sort((a, b) => b.localeCompare(a)));
+      } else {
+        setGoodsArr(prevState => [...prevState].sort((a, b) => a.localeCompare(b)));
+      }
+
+      setSortValue(SORT_BY_ALPHABETICALLY);
+      setResetValue(true);
+
+      return;
+    }
+
+    if (value === SORT_BY_LENGTH) {
+      if (reverseValue) {
+        setGoodsArr(prevState => [...prevState].sort((a, b) => b.length - a.length));
+      } else {
+        setGoodsArr(prevState => [...prevState].sort((a, b) => a.length - b.length));
+      }
+
+      setSortValue(SORT_BY_LENGTH);
+      setResetValue(true);
+
+      return;
+    }
+
+    setGoodsArr(goodsFromServer);
+    setReverseValue(false);
+    setResetValue(false);
+    setSortValue('');
+  };
+
+  const reverse = () => {
+    if (!sortValue) {
+      setResetValue(!resetValue);
+    } else {
+      setResetValue(true);
+    }
+
+    setGoodsArr(prevState => [...prevState].reverse());
+    setReverseValue(!reverseValue);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortValue !== SORT_BY_ALPHABETICALLY,
+          })}
+          onClick={() => sortBy(SORT_BY_ALPHABETICALLY)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortValue !== SORT_BY_LENGTH,
+          })}
+          onClick={() => sortBy(SORT_BY_LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': !reverseValue,
+          })}
+          onClick={reverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {resetValue && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => sortBy('')}
```

Comment:
this is have no logic for reset state you call func sortBy, please create separate func for this logic and remove this logic from sortBy